pkg/server/infrastructure/datastore/opengauss/opengauss.go

```diff
@@ -0,0 +1,320 @@
+/*
+Copyright 2021 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package opengauss
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	postgresorm "gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+	"gorm.io/gorm/logger"
+	"k8s.io/klog/v2"
+
+	"github.com/kubevela/velaux/pkg/server/domain/model"
+	"github.com/kubevela/velaux/pkg/server/infrastructure/datastore"
+	"github.com/kubevela/velaux/pkg/server/infrastructure/datastore/sqlnamer"
+)
+
+type opengauss struct {
+	client   gorm.DB
+	database string
+}
+
+// New opengauss datastore instance
+func New(ctx context.Context, cfg datastore.Config) (datastore.DataStore, error) {
+
+	db, err := gorm.Open(postgresorm.Open(cfg.URL), &gorm.Config{
+		NamingStrategy:                           sqlnamer.SQLNamer{},
+		DisableForeignKeyConstraintWhenMigrating: true,
+		Logger:                                   logger.Default.LogMode(logger.Silent),
```

Comment:
Is it ok to set logger level to `Silent`?

Reply:
I changed it to an Error level